_posts/2015-10-29-code-talks-java.md

```diff
@@ -0,0 +1,72 @@
+Things to keep in mind though: All used `npm` packages must be "nashorn compatible".
+Whatever this means.
+
+For reading this far you deserve a well kept secret about `npm`: It's **not** an abbreviation for the **n**ode **p**ackage **m**anager but a recursive acronymic abbreviation for "npm is not an acronym".
+
+However rumor has it, the package manager sends "**n**aughty **p**ush **m**essages" once installed (see [npm-expansions](https://github.com/npm/npm-expansions)).
+
+# 2. We're pretty much up-to-date
+
+In the Frontend/JavaScript track there were several talks covering various aspects.
+From the language itself ('There is no JavScript' - awesome presentation by Noam Kfir) to pretty interesting testing approaches and the life of suffering with React and ES6.
+We realised that our current testing and build pipeline which consists of `mocha`, `babel`, `karma` and `webpack` is not only considered state of the art but is just discovered by some companies.
+They shared knowledge of pitfalls they encountered and how they overcame them (or at least what they tried to do so).
+Their solution was basically our current build configuration.
+But since the topic is pretty tricky, there were of course some nifty tricks we've yet to adopt so we're very glad they went through all that pain and shared these valuable findings.
+Anyway it still gave us a pretty confident feeling about how we're doing things.
+In these days it seems to be the mix of Es6, Es5 and React modules together with universal JavaScript that is the most challenging thing to overcome.
+
+# 3. Operations is your new best friend
+
+First I have to say that this talk, done by two guys from 1&1, was really inspiring and made you think of changing a thing or two in you own development workflow.
+
+Plainly put, the whole talk was about **DevOps** and how this construct can help to smoothen relations between all kinds of different departments throughout the whole company.
+Translated to ourselves they mainly spoke of improving workflows between the RND department (the developers) and the team which takes care of managing and deploying the application (AM).
+The scenario described was one that I myself experienced one time or another: The over-the-fence phenomenon.
+A developer finishes a feature and sends it to Operations.
+Now they can deal with it!
+**But**: It might be hard to deploy because of several reasons (be it a changed database structure or some changes in the build pipeline), it might have bugs, break other features or cause who knows what.
+So a developer should not only take care of finishing the feature assigned to him but also have in mind that there's work *after* he did his final commit.
+He (or she of course) should feel responsible for the software as a whole (and everything that is related to it).
+
+The 1&1 guys gave several tips how to improve communications and workflows and also talked about how *they* tried to establish DevOps.
```

Comment:
The 1&1 guys gave several tips on how to improve communications and workflows. They also talked about how *they* tried to establish DevOps.